pkg/controller/drain.go

```diff
@@ -313,39 +338,42 @@ func (o *DrainOptions) deleteOrEvictPods(pods []api.Pod) error {
 	return o.deletePods(pods, getPodFn)
 }
 
+func volIsPvc(vol *corev1.Volume) bool {
+	return vol.PersistentVolumeClaim != nil
+}
+
+func filterPodsWithPv(pods []api.Pod) ([]*api.Pod, []*api.Pod) {
+	podsWithPv, podsWithoutPv := []*api.Pod{}, []*api.Pod{}
+
+	for i := range pods {
+		hasPv := false
+		pod := &pods[i]
+		vols := pod.Spec.Volumes
+		for k := range vols {
+			vol := &vols[k]
+			hasPv = volIsPvc(vol)
+			if hasPv {
+				podsWithPv = append(podsWithPv, pod)
+				// No need to process rest of the volumes
+				break
+			}
+		}
+		if !hasPv {
+			podsWithoutPv = append(podsWithoutPv, pod)
+		}
+	}
+	return podsWithPv, podsWithoutPv
+}
+
 func (o *DrainOptions) evictPods(pods []api.Pod, policyGroupVersion string, getPodFn func(namespace, name string) (*api.Pod, error)) error {
 	returnCh := make(chan error, 1)
```

Comment:
```suggestion
	returnCh := make(chan error, 1)
	defer close(returnCh)
```

Reply:
If this channel is closed before all the go routines are done writing to this channel, it could lead to panic. I suggest we leave this channel while returning, and let it be taken care of by the GC, as is done [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/drain/drain.go#L354) from where most of the drain logic is imported:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/drain/drain.go#L354

WDYT?

Reply:
Also, as mentioned in [The Go Programming Language](https://www.gopl.io/):
> You needn't close every channel when you've finished with it. It's only necessary to close a channel when it is important to tell the receiving goroutines that all data have been sent. A channel that the garbage collector determinies to be unreachable will have its resources reclaimed whether or not it is closed.

Reply:
It might be ok to not close the channel, but the point is why would any go routines be running after the outer func has returned? Shouldn't we make sure they don't? That is the actual review comment. Just like the last time ;-)